### PR TITLE
add filesystem checks

### PR DESCRIPTION
**Pull request by amdprophet**
Signed-off-by: Justin Kolberg <amd.prophet@gmail.com>

### CODE REVIEW

File: system/filesystem/check-checksums.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-checksums
+  namespace: default
+spec:
+  command: check-checksums.rb -f /bin/sh -h eca410732dd8568ec169c1879965f4361bd69d72c06cb5990c7b3962f197a844
```
Comment (nikkictl) [inline system/filesystem/check-checksums.yml]: What does -h stand for here? Could we replace this value with a label?

File: system/filesystem/check-dir-count.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-dir-count
+  namespace: default
+spec:
+  command: check-dir-count.rb -d /tmp -w 75 -c 100
```
Comment (nikkictl) [inline system/filesystem/check-dir-count.yml]:
```suggestion
  command: check-dir-count.rb -d /tmp -w {{ .labels.check_dir_warning | default 75 }} -c {{ .labels.check_dir_critical | default 100 }}
```

File: system/filesystem/check-dir-size.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-dir-size
+  namespace: default
+spec:
+  command: check-dir-size.rb -d /tmp -w 1000000000 -c 2000000000
```
Comment (nikkictl) [inline system/filesystem/check-dir-size.yml]:
```suggestion
  command: check-dir-size.rb -d /tmp -w {{ .labels.check_dir_size_warning -c {{ .labels.check_dir_size_critical }}
```

File: system/filesystem/check-file-size.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-file-size
+  namespace: default
+spec:
+  command: check-file-size.rb -f /bin/sh -w 2000000 -c 3000000
```
Comment (nikkictl) [inline system/filesystem/check-file-size.yml]:
```suggestion
  command: check-file-size.rb -f /bin/sh -w {{ .labels.check_file_size_warning | default 2000000 }} -c {{ .labels.check_file_size_critical | default 3000000 }}
```

File: system/filesystem/check-mtime.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-mtime
+  namespace: default
+spec:
+  command: check-mtime.rb -f /bin/sh -w 3600 -c 7200
```
Comment (nikkictl) [inline system/filesystem/check-mtime.yml]:
```suggestion
  command: check-mtime.rb -f /bin/sh -w {{ .labels.check_mtime_warning | default 3600 }} -c {{ .labels.check_mtime_critical | default 7200 }}
```

File: system/filesystem/check-checksums.yml (hunk 1 above)
Comment (amdprophet) [reply system/filesystem/check-checksums.yml]: It's the sha256 checksum of `/bin/sh` and both values should probably be replaced with labels.

File: system/filesystem/check-tail.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-tail
+  namespace: default
+spec:
+  command: check-tail.rb -f /var/log/messages -P error -l 10
```
Comment (nikkictl) [inline system/filesystem/check-tail.yml]: Not sure what -l stands for, but more labels with defaults like in the other examples would be a good approach!

File: system/filesystem/check-tail.yml [hunk 7]
```diff
@@ -0,0 +1,105 @@
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: check-tail
+  namespace: default
+spec:
+  command: check-tail.rb -f {{ .labels.check_tail_fail | default /var/log/messages }} -P {{ .labels.check_tail_pattern | default error }} -l {{ .labels.check_tail_lines | default 10 }}
```
Comment (nikkictl) [inline system/filesystem/check-tail.yml]: tail_fail made me giggle 🤣

**Review by nikkictl — APPROVED** [review]
Thanks for updating to labels with defaults!

File: system/filesystem/check-tail.yml (hunk 7 above)
Comment (amdprophet) [reply system/filesystem/check-tail.yml]: Hah - that was a typo! It should be file. I'll update this.

**Pull request merged**